File: source/cpBase/CPHtmlBaseClass.vb
'
' documentation should be in a new project that inherits these classes. The class names should be the object names in the actual cp project
'
Namespace Contensive.BaseClasses
    Public MustInherit Class CPHtmlBaseClass

        Enum EditorContentScope
            Page = 1
            Email = 2
            PageTemplate = 3
        End Enum
        '
        Enum EditorUserScope
            Developer = 1
            Administrator = 2
            ContentManager = 3
            PublicUser = 4
            CurrentUser = 5
        End Enum
        'Public Sub New(ByVal cmcObj As Contensive.Core.cpCoreClass, ByRef CPParent As CPBaseClass)
        Public MustOverride Function div(ByVal InnerHtml As String, Optional ByVal HtmlName As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.div
        Public MustOverride Function p(ByVal InnerHtml As String, Optional ByVal HtmlName As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.p
        Public MustOverride Function li(ByVal InnerHtml As String, Optional ByVal HtmlName As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.li
        Public MustOverride Function ul(ByVal InnerHtml As String, Optional ByVal HtmlName As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.ul
        Public MustOverride Function ol(ByVal InnerHtml As String, Optional ByVal HtmlName As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.ol
        Public MustOverride Function CheckBox(ByVal HtmlName As String, Optional ByVal HtmlValue As Boolean = False, Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.CheckBox
        Public MustOverride Function CheckList(ByVal HtmlName As String, ByVal PrimaryContentName As String, ByVal PrimaryRecordId As Integer, ByVal SecondaryContentName As String, ByVal RulesContentName As String, ByVal RulesPrimaryFieldname As String, ByVal RulesSecondaryFieldName As String, Optional ByVal SecondaryContentSelectSQLCriteria As String = "", Optional ByVal CaptionFieldName As String = "", Optional ByVal IsReadOnly As Boolean = False, Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.CheckList
        Public MustOverride Function Form(ByVal InnerHtml As String, Optional ByVal HtmlName As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "", Optional ByVal ActionQueryString As String = "", Optional ByVal Method As String = "post") As String 'Implements BaseClasses.CPHtmlBaseClass.Form
        Public MustOverride Function h1(ByVal InnerHtml As String, Optional ByVal HtmlName As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.h1
        Public MustOverride Function h2(ByVal InnerHtml As String, Optional ByVal HtmlName As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.h2
        Public MustOverride Function h3(ByVal InnerHtml As String, Optional ByVal HtmlName As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.h3
        Public MustOverride Function h4(ByVal InnerHtml As String, Optional ByVal HtmlName As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.h4
        Public MustOverride Function h5(ByVal InnerHtml As String, Optional ByVal HtmlName As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.h5
        Public MustOverride Function h6(ByVal InnerHtml As String, Optional ByVal HtmlName As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.h6
        Public MustOverride Function RadioBox(ByVal HtmlName As String, ByVal HtmlValue As String, ByVal CurrentValue As String, Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.RadioBox
        Public MustOverride Function SelectContent(ByVal HtmlName As String, ByVal HtmlValue As String, ByVal ContentName As String, Optional ByVal SQLCriteria As String = "", Optional ByVal NoneCaption As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.SelectContent
        Public MustOverride Function SelectList(ByVal HtmlName As String, ByVal HtmlValue As String, ByVal OptionList As String, Optional ByVal NoneCaption As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.SelectList
        Public MustOverride Sub ProcessCheckList(ByVal HtmlName As String, ByVal PrimaryContentName As String, ByVal PrimaryRecordID As String, ByVal SecondaryContentName As String, ByVal RulesContentName As String, ByVal RulesPrimaryFieldname As String, ByVal RulesSecondaryFieldName As String) 'Implements BaseClasses.CPHtmlBaseClass.ProcessCheckList
        Public MustOverride Sub ProcessInputFile(ByVal HtmlName As String, Optional ByVal VirtualFilePath As String = "") 'Implements BaseClasses.CPHtmlBaseClass.ProcessInputFile
        Public MustOverride Function Hidden(ByVal HtmlName As String, ByVal HtmlValue As String, Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.Hidden
        Public MustOverride Function InputDate(ByVal HtmlName As String, Optional ByVal HtmlValue As String = "", Optional ByVal Width As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.InputDate
        Public MustOverride Function InputFile(ByVal HtmlName As String, Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.InputFile
        Public MustOverride Function InputText(ByVal HtmlName As String, Optional ByVal HtmlValue As String = "", Optional ByVal Height As String = "", Optional ByVal Width As String = "", Optional ByVal IsPassword As Boolean = False, Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.InputText
        Public MustOverride Function InputTextExpandable(ByVal HtmlName As String, Optional ByVal HtmlValue As String = "", Optional ByVal Rows As Integer = 0, Optional ByVal StyleWidth As String = "", Optional ByVal IsPassword As Boolean = False, Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.InputTextExpandable
        Public MustOverride Function SelectUser(ByVal HtmlName As String, ByVal HtmlValue As Integer, ByVal GroupId As Integer, Optional ByVal NoneCaption As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.SelectUser
        Public MustOverride Function Indent(ByVal SourceHtml As String, Optional ByVal TabCnt As Integer = 1) As String 'Implements BaseClasses.CPHtmlBaseClass.Indent
        Public MustOverride Function InputWysiwyg(ByVal HtmlName As String, Optional ByVal HtmlValue As String = "", Optional ByVal UserScope As BaseClasses.CPHtmlBaseClass.EditorUserScope = BaseClasses.CPHtmlBaseClass.EditorUserScope.CurrentUser, Optional ByVal ContentScope As BaseClasses.CPHtmlBaseClass.EditorContentScope = BaseClasses.CPHtmlBaseClass.EditorContentScope.Page, Optional ByVal Height As String = "", Optional ByVal Width As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String 'Implements BaseClasses.CPHtmlBaseClass.InputWysiwyg
        Public MustOverride Sub AddEvent(ByVal HtmlId As String, ByVal DOMEvent As String, ByVal JavaScript As String)
        Public MustOverride Function Button(ByVal HtmlName As String, Optional ByVal HtmlValue As String = "", Optional ByVal HtmlClass As String = "", Optional ByVal HtmlId As String = "") As String
        Public MustOverride Function adminHint(ByVal innerHtml As String) As String
    End Class

End Namespace

File: source/CPBase50/CPHtmlBaseClass.cs
//========================================================================
// This conversion was produced by the Free Edition of
// Instant C# courtesy of Tangible Software Solutions.
// Order the Premium Edition at https://www.tangiblesoftwaresolutions.com
//========================================================================

//
// documentation should be in a new project that inherits these classes. The class names should be the object names in the actual cp project
//

//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Diagnostics;

namespace Contensive.BaseClasses
{
	public abstract class CPHtmlBaseClass
	{

		public enum EditorContentScope
		{
			Page = 1,
			Email = 2,
			PageTemplate = 3
		}
		//
		public enum EditorUserScope
		{
			Developer = 1,
			Administrator = 2,
			ContentManager = 3,
			PublicUser = 4,
			CurrentUser = 5
		}
		//Public Sub New(ByVal cmcObj As Contensive.Core.cpCoreClass, ByRef CPParent As CPBaseClass)
		public abstract string div(string InnerHtml, string HtmlName = "", string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.div
		public abstract string p(string InnerHtml, string HtmlName = "", string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.p
		public abstract string li(string InnerHtml, string HtmlName = "", string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.li
		public abstract string ul(string InnerHtml, string HtmlName = "", string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.ul
		public abstract string ol(string InnerHtml, string HtmlName = "", string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.ol
		public abstract string CheckBox(string HtmlName, bool HtmlValue = false, string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.CheckBox
		public abstract string CheckList(string HtmlName, string PrimaryContentName, int PrimaryRecordId, string SecondaryContentName, string RulesContentName, string RulesPrimaryFieldname, string RulesSecondaryFieldName, string SecondaryContentSelectSQLCriteria = "", string CaptionFieldName = "", bool IsReadOnly = false, string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.CheckList
		public abstract string Form(string InnerHtml, string HtmlName = "", string HtmlClass = "", string HtmlId = "", string ActionQueryString = "", string Method = "post"); //Implements BaseClasses.CPHtmlBaseClass.Form
		public abstract string h1(string InnerHtml, string HtmlName = "", string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.h1
		public abstract string h2(string InnerHtml, string HtmlName = "", string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.h2
		public abstract string h3(string InnerHtml, string HtmlName = "", string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.h3
		public abstract string h4(string InnerHtml, string HtmlName = "", string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.h4
		public abstract string h5(string InnerHtml, string HtmlName = "", string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.h5
		public abstract string h6(string InnerHtml, string HtmlName = "", string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.h6
		public abstract string RadioBox(string HtmlName, string HtmlValue, string CurrentValue, string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.RadioBox
		public abstract string SelectContent(string HtmlName, string HtmlValue, string ContentName, string SQLCriteria = "", string NoneCaption = "", string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.SelectContent
		public abstract string SelectList(string HtmlName, string HtmlValue, string OptionList, string NoneCaption = "", string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.SelectList
		public abstract void ProcessCheckList(string HtmlName, string PrimaryContentName, string PrimaryRecordID, string SecondaryContentName, string RulesContentName, string RulesPrimaryFieldname, string RulesSecondaryFieldName); //Implements BaseClasses.CPHtmlBaseClass.ProcessCheckList
		public abstract void ProcessInputFile(string HtmlName, string VirtualFilePath = "");

		// no -- add these to the filesystem object
		//Public MustOverride Function ProcessInputFile(ByVal HtmlName As String, fileSystem As CPFileSystemBaseClass, ByRef returnPathFilename As String) As Boolean
		//Public MustOverride Function ProcessInputFile(ByVal HtmlName As String, fileSystem As CPFileSystemBaseClass, uploadFilePath As String, ByRef returnFilename As String) As Boolean

		public abstract string Hidden(string HtmlName, string HtmlValue, string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.Hidden
		public abstract string InputDate(string HtmlName, string HtmlValue = "", string Width = "", string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.InputDate
		public abstract string InputFile(string HtmlName, string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.InputFile
		public abstract string InputText(string HtmlName, string HtmlValue = "", string Height = "", string Width = "", bool IsPassword = false, string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.InputText
		public abstract string InputTextExpandable(string HtmlName, string HtmlValue = "", int Rows = 0, string StyleWidth = "", bool IsPassword = false, string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.InputTextExpandable
		public abstract string SelectUser(string HtmlName, int HtmlValue, int GroupId, string NoneCaption = "", string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.SelectUser
		public abstract string Indent(string SourceHtml, int TabCnt = 1); //Implements BaseClasses.CPHtmlBaseClass.Indent
		public abstract string InputWysiwyg(string HtmlName, string HtmlValue = "", BaseClasses.CPHtmlBaseClass.EditorUserScope UserScope = BaseClasses.CPHtmlBaseClass.EditorUserScope.CurrentUser, BaseClasses.CPHtmlBaseClass.EditorContentScope ContentScope = BaseClasses.CPHtmlBaseClass.EditorContentScope.Page, string Height = "", string Width = "", string HtmlClass = "", string HtmlId = ""); //Implements BaseClasses.CPHtmlBaseClass.InputWysiwyg
		public abstract void AddEvent(string HtmlId, string DOMEvent, string JavaScript);
		public abstract string Button(string HtmlName, string HtmlValue = "", string HtmlClass = "", string HtmlId = "");
		public abstract string adminHint(string innerHtml);
	}

}

